common/test/acceptance/tests/lms/test_lms_edxnotes.py

```diff
@@ -1005,6 +1063,290 @@ def test_open_note_when_accessed_from_notes_page(self):
+            next_button_enabled=False,
+            current_page_number=2,
+            total_pages=2
+        )
+
+    def test_search_behaves_correctly_with_pagination(self):
+        """
+        Scenario: Searching behaves correctly with pagination.
+
+        Given that I am a registered user
+        And I have a course with 12 notes
+        When I open Notes page
+        Then I can see notes list contains 10 items
+        When I run the search with "note" query
+        Then I see no error message
+        And I see that "Search Results" tab appears with 11 notes found
+        And an event has fired indicating that the Search Results view was selected
+        And an event has fired recording the search that was performed
+        """
+        self.search_and_verify()
+
+        self._verify_pagination_info(
+            notes_count_on_current_page=10,
+            header_text='Showing 1-10 out of 11 total',
+            previous_button_enabled=False,
+            next_button_enabled=True,
+            current_page_number=1,
+            total_pages=2
+        )
+
+        self.assert_viewed_event('Search Results')
+        self.assert_search_event('note', 11)
+
+    def test_search_with_next_and_prev_page_button(self):
+        """
+        Scenario: Next & Previous buttons are working as expected for search
+
+        Given that I am a registered user
+        And I have a course with 12 notes
+        When I open Notes page
+        Then I can see notes list with 10 items
+        And I should see paging header and footer with correct data
+        And previous button is disabled
+        And next button is enabled
+
+        When I run the search with "note" query
+        Then I see that "Search Results" tab appears with 11 notes found
+        And an event has fired indicating that the Search Results view was selected
+        And an event has fired recording the search that was performed
+
+        When I click on next page button in footer
+        Then I should be navigated to second page
+        And I should see a list with 1 item
+        And I should see paging header and footer with correct info
+        And I should see enabled previous button
+        And I should also see disabled next button
+
+        When I click on previous page button in footer
+        Then I should be navigated to first page
+        And I should see a list with 10 items
+        And I should see paging header and footer with correct info
+        And I should see disabled previous button
+        And I should also see enabled next button
+        """
+        self.search_and_verify()
+
+        self._verify_pagination_info(
+            notes_count_on_current_page=10,
+            header_text='Showing 1-10 out of 11 total',
+            previous_button_enabled=False,
+            next_button_enabled=True,
+            current_page_number=1,
+            total_pages=2
+        )
+
+        self.assert_viewed_event('Search Results')
```

Comment:
Currently it is only being triggered on first page. I am not sure about the requirements as edxnotes event document has no information regarding events behavior in pagination.